.github/workflows/test_profile.yml

```diff
@@ -0,0 +1,96 @@
+name: Test the CIS M365 Profile
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  validate:
+    name: Validate my profile
+    runs-on: ubuntu-latest
+    env:
+      CHEF_LICENSE: accept-silent
+      CHEF_LICENSE_KEY: ${{ secrets.SAF_CHEF_LICENSE_KEY }}
+      PLATFORM: "m365"
+      LC_ALL: "en_US.UTF-8"
+    steps:
+      - name: Add needed packages
+        run: |
+          sudo apt-get update
+          sudo apt-get -y install jq
+
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Set short git commit SHA
+        id: vars
+        run: |
+          calculatedSha=$(git rev-parse --short ${{ github.sha }})
+          echo "COMMIT_SHORT_SHA=$calculatedSha" >> $GITHUB_ENV
+
+      - name: Confirm git commit SHA output
+        run: echo ${{ env.COMMIT_SHORT_SHA }}
+
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "3.1"
+
+      - name: Disable ri and rdoc
+        run: 'echo "gem: --no-ri --no-rdoc" >> ~/.gemrc'
+
+      - name: Run Bundle Install
+        run: bundle install
+
+      - name: Check Inspec version
+        run: bundle exec inspec version
+
+      - name: Vendor the Profile
+        run: bundle exec inspec vendor . --overwrite
+
+      - name: Collect certificate file
+        shell: pwsh
+        run: |
+          "[IO.File]::WriteAllBytes(cert.pfx,(${{secrets.SAF_365_CERTIFICATE}} | ConvertFrom-Json | %{[byte]$_}))"
+
+      - name: Run Inspec test
+        continue-on-error: true
+        run: |
+          bundle exec inspec exec . \
+          --controls=microsoft-365-foundations-2.1.5 \
```

Comment:
We should figure out how to make this a variable we can pass in via the commit message if we want to be able to do this kind of thing and it should have logic in it to only add the --controls flag if the commit message has that variable